x-pack/plugins/canvas/public/expression_types/datasources/esdocs.js

```diff
@@ -54,48 +55,140 @@ const EsdocsDatasource = ({ args, updateArgs }) => {
+          <FormattedMessage
+            id="xpack.canvas.expressionTypes.datasource.formRowIndexHelpText"
+            defaultMessage="Enter an index name or select an index pattern"
+          />
+        }
+      >
         <ESIndexSelect value={index} onChange={index => setArg('index', index)} />
       </EuiFormRow>
 
-      <EuiFormRow label="Query" helpText="Lucene query string syntax" compressed>
+      <EuiFormRow
+        label={
+          <FormattedMessage
+            id="xpack.canvas.expressionTypes.datasource.formRowQueryLabel"
+            defaultMessage="Query"
+          />
+        }
+        helpText={
+          <FormattedMessage
+            id="xpack.canvas.expressionTypes.datasource.formRowQueryHelpText"
+            defaultMessage="Lucene query string syntax"
+          />
+        }
+        compressed
+      >
         <EuiFieldText value={getQuery()} onChange={e => setArg(getArgName(), e.target.value)} />
       </EuiFormRow>
-      <EuiFormRow label="Sort Field" helpText="Document sort field">
+      <EuiFormRow
+        label={
+          <FormattedMessage
+            id="xpack.canvas.expressionTypes.datasource.formRowSortFieldLabel"
+            defaultMessage="Sort Field"
+          />
+        }
+        helpText={
+          <FormattedMessage
+            id="xpack.canvas.expressionTypes.datasource.formRowSortFieldHelpText"
+            defaultMessage="Document sort field"
+          />
+        }
+      >
         <ESFieldSelect
           index={index}
           value={sortField}
           onChange={field => setArg('sort', [field, sortOrder].join(', '))}
         />
       </EuiFormRow>
-      <EuiFormRow label="Sort Order" helpText="Document sort order" compressed>
+      <EuiFormRow
+        label={
+          <FormattedMessage
+            id="xpack.canvas.expressionTypes.datasource.formRowSortOrderLabel"
+            defaultMessage="Sort Order"
+          />
+        }
+        helpText={
+          <FormattedMessage
+            id="xpack.canvas.expressionTypes.datasource.formRowSortOrderHelpText"
+            defaultMessage="Document sort order"
+          />
+        }
+        compressed
+      >
         <EuiSelect
           value={sortOrder.toLowerCase()}
           onChange={e => setArg('sort', [sortField, e.target.value].join(', '))}
           options={sortOptions}
         />
       </EuiFormRow>
       <EuiFormRow
-        label="Fields"
+        label={
+          <FormattedMessage
+            id="xpack.canvas.expressionTypes.datasource.formRowFieldsLabel"
+            defaultMessage="Fields"
+          />
+        }
         helpText={
-          fields.length <= 10
-            ? 'The fields to extract. Kibana scripted fields are not currently available'
-            : 'This datasource performs best with 10 or fewer fields'
+          fields.length <= 10 ? (
+            <FormattedMessage
+              id="xpack.canvas.expressionTypes.datasource.formRowFields.fieldsLessThen10HelpText"
```

Comment:
I'm not sure it is good idea to use then10, in future it can be 20 or 30 and we will need to update id